napari/components/dims.py

```diff
@@ -67,285 +66,271 @@ class Dims(EventedModel):
+            )
+        ]
+
+    @property
+    def nsteps(self) -> List[float]:
+        return [
+            int((max_val - min_val) // step)
+            for (min_val, max_val), step in zip(self.range, self.step)
+        ]
+
+    @nsteps.setter
+    def nsteps(self, value):
+        self.step = [
+            (max_val - min_val) / nsteps
+            for (min_val, max_val), nsteps in zip(self.range, value)
+        ]
+
+    @property
+    def thickness(self) -> List[float]:
+        return [high - low for low, high in self.span]
+
+    @thickness.setter
+    def thickness(self, value):
+        span = []
+        for thickness, (min_val, max_val), (low, high) in zip(
+            value, self.range, self.span
+        ):
+            max_change = min(abs(low - min_val), abs(max_val - high))
+            thickness_change = min((thickness - (high - low)) / 2, max_change)
+            new_low = max(min_val, low - thickness_change)
+            new_high = min(max_val, high + thickness_change)
+            span.append([new_low, new_high])
+        self.span = span
+
+    @property
+    def _thickness_step(self) -> List[float]:
+        return [
+            thickness / step
+            for thickness, step in zip(self.thickness, self.step)
+        ]
+
+    @_thickness_step.setter
+    def _thickness_step(self, value):
+        self.thickness = [
+            thickness * step for thickness, step in zip(value, self.step)
+        ]
+
+    @property
+    def point(self) -> List[float]:
+        return [(low + high) / 2 for low, high in self.span]
+
+    @point.setter
+    def point(self, value):
+        span = []
+        for point, (min_val, max_val), thickness in zip(
+            value, self.range, self.thickness
+        ):
+            half_thk = thickness / 2
+            min_pt, max_pt = (min_val + half_thk, max_val - half_thk)
+            point = np.clip(point, min_pt, max_pt)
+            span.append([point - half_thk, point + half_thk])
+        self.span = span
 
     @property
-    def point(self) -> Tuple[int, ...]:
-        """Tuple of float: Value of each dimension."""
-        # The point value is computed from the range and current_step
-        point = tuple(
-            min_val + step_size * value
-            for (min_val, max_val, step_size), value in zip(
-                self.range, self.current_step
+    def _point_step(self):
+        return [
+            int(round((point - min_val) / step))
+            for point, (min_val, _), step in zip(
+                self.point, self.range, self.step
             )
+        ]
+
+    @_point_step.setter
+    def _point_step(self, value):
+        self.point = [
+            min_val + point * step
+            for point, (min_val, _), step in zip(value, self.range, self.step)
+        ]
+
+    @property
+    def current_step(self) -> List[int]:
+        warnings.warn(
+            trans._(
+                'Dims.current_step is deprecated. Use Dims._point_step instead.'
```

Comment:
Was this necessary? It looks like we use `dims._point_step` a lot in our tests? In general if deprecating something we shouldn't point people towards a private property. Does `point_step` need to be public instead?

Reply:
Maybe also we could do deprecation in a follow-up PR - it might make the diff smaller and this PR easier to review

Reply:
I forgot to mention in the post above: nothing about naming/deprecations should be taken as "final", even all the private/public choices are somewhat random at this point. Definitely should go through them before any merge to make sure they make sense :)